source/VirtualDesktop/ApplicationHelper.cs

```diff
@@ -9,16 +9,29 @@ public static class ApplicationHelper
 	{
 		internal static IApplicationView GetApplicationView(this IntPtr hWnd)
 		{
-			IApplicationView view;
-			ComObjects.ApplicationViewCollection.GetViewForHwnd(hWnd, out view);
-
-			return view;
+			try
+			{
+				IApplicationView view;
+				ComObjects.ApplicationViewCollection.GetViewForHwnd(hWnd, out view);
+				return view;
+			}
+			catch (System.Runtime.InteropServices.COMException ex) when (ex.Match(HResult.TYPE_E_ELEMENTNOTFOUND))
+			{
+				return null;
+			}
 		}
 
 		public static string GetAppId(IntPtr hWnd)
 		{
+			var view = hWnd.GetApplicationView();
+
+			if (view == null)
+			{
+				throw new ArgumentException(nameof(hWnd));
```

Comment:
this should message out why its an argument exception, and probably not use C#6.0 for more broad compilation possibility:
```throw new ArgumentException("hWnd specified not found","hWnd");```